src/main/java/seedu/duke/Save.java

```diff
@@ -1,37 +1,42 @@
 package seedu.duke;
 
-
 import seedu.duke.ai.Ai;
 import seedu.duke.stats.MatchStat;
 import seedu.duke.ui.Ui;
 
-
 public class Save {
 
-
     public static void executeSave() {
         int userSaveDirection = getUserSaveDirection();
         int aiPenaltyDirection = Ai.getAiDirection();
         boolean isGoalSaved = userSaveDirection == aiPenaltyDirection;
-
-
         Formatter.printSaveResult(isGoalSaved);
         MatchStat.updateStat(isGoalSaved);
     }
 
-
     private static int getUserSaveDirection() {
-        int userDirection;
-        do {
+        int userDirection = -1;
+        while (userDirection == -1) {
             System.out.print("Enter the direction to save the penalty (0, 1, or 2): ");
             String directionString = Ui.IN.nextLine().trim();
```

Comment:
You are reading save int right?

Reply:
Yes. The getUserSaveDirection() method prompts the user for input and expects an integer corresponding to the direction to save the penalty. The input is read as a string and then converted into an integer using the convertToShootDirection() method, which checks if the input is within the valid range (0, 1, or 2).